src/server/game/Entities/Object/Object.cpp

```diff
@@ -2739,6 +2739,30 @@ Position WorldObject::GetRandomNearPosition(float radius)
     return pos;
 }
 
+Position WorldObject::GetRandomNearPositionWithCollisionCheck(Unit* caster, float radius)
+{
+    Position pos = GetRandomNearPosition(radius);
+
+    for (float angle = float(M_PI) / 8; angle < float(M_PI) * 2; angle += float(M_PI) / 8)
+    {
+        Position endPos = pos;
+        endPos.RelocatePolarOffset(angle, radius);
+
+        G3D::Vector3 hitResult;
+
+        if (caster->GetMap()->GetObjectHitPos(caster->GetPhaseMask(), pos.m_positionX, pos.m_positionY, pos.m_positionZ,
+            endPos.m_positionX, endPos.m_positionY, endPos.m_positionZ,
+            hitResult.x, hitResult.y, hitResult.z, 0.0f))
+        {
+            pos.Relocate(hitResult.x, hitResult.y, hitResult.z);
+            pos.RelocatePolarOffset(angle + 180.f, radius);
+            break;
+        }
+    }
+
+    return pos;
+}
+
```

Comment:
Yes the commit is incorrect. I revert it but I'm stuck because GetObjectHitPos doesn't detect walls...

Reply:
From my understanding, your current solution (aside from the problem with GetObjectHitResult), is almost ignoring the player's position, and in the provided example, I pointed out an issue that you may encounter.

So I still think that:
1. Is position of the caster.
2. Possible random point near caster (`pos` variable).
3. Is one of 8 ray casts which hits the wall.
4. Final position.

Reply:
I'm not sure to understand. GerRandomNearPosition take the caster position and randomize a new one with radius define in the method. It will also check that the new position is valid.

Reply:
Yep, I understand that. I just wanted to point out how your algorithm would behave if GetRandomNearPosition returned a point that was on the other side of a wall from the player.

Reply:
Oh I understand, I forget that Acore::IsValidMapCoord doesn't check LoS case that a position outside of a wall stay a position valid on map.